src/openforms/js/components/form/edit/options.js

```diff
@@ -208,6 +208,7 @@ const OPTIONS_CHOICES = [
         },
       },
     ],
+    clearOnHide: true,
```

Comment:
I don't believe this contributed anything to breaking the infinite change event loop, but it seems more correct either way.